### PR TITLE
api: return not found when region doesn't exist

### LINKED ISSUE #8868: pd-ctl return error result when region doesn't exist
### What did you do?
query a region by id

### What did you expect to see?
Image

### What did you see instead?
Image

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #8868

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
Before:
image
After:
image

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/api/region.go [hunk 1]
```diff
@@ -99,6 +103,10 @@ func (h *regionHandler) GetRegion(w http.ResponseWriter, r *http.Request) {
 	}
 
 	regionInfo := rc.GetRegionByKey(paramsByte[0])
+	if regionInfo == nil {
+		h.rd.JSON(w, http.StatusNotFound, errs.ErrRegionNotFound.FastGenByArgs().Error())
```
Comment (rleungx) [inline server/api/region.go:107]: 404 is expected?

Comment (lhy1024) [reply server/api/region.go:107]: Which status code do you think is more reasonable?

**Comment by lhy1024** [comment]
tidb, tikv, tiup do not call `region/key` or `region/id`

File: server/api/region.go (hunk 1 above)
Comment (rleungx) [reply server/api/region.go:107]: I was wondering if a 0 is valid request.

**Review by JmPotato — COMMENTED** [review]
My concern here is that some of the downstream components may rely on this behavior to retry (when the result is empty), will this break the compatibility?

**Comment by lhy1024** [comment]
> My concern here is that some of the downstream components may rely on this behavior to retry (when the result is empty), will this break the compatibility?

this api is not used by tidb, tikv, tiup, it is used by pd-ctl

File: server/api/region.go (hunk 1 above)
Comment (lhy1024) [reply server/api/region.go:107]: why is 0 valid request?

Comment (rleungx) [reply server/api/region.go:107]: 0 is invalid, but here you return 404 instead of 400.

Comment (lhy1024) [reply server/api/region.go:111]: got it

**Review by rleungx — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**